ibm/service/power/resource_ibm_pi_host.go

```diff
@@ -0,0 +1,380 @@
+			},
+		},
+	}
+}
+
+func resourceIBMPIHostCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	client := instance.NewIBMPIHostGroupsClient(ctx, sess, cloudInstanceID)
+	hosts := d.Get(Arg_Host).([]interface{})
+	hostgroupID := d.Get(Arg_HostgroupID).(string)
+	body := models.HostCreate{}
+	hostBody := make([]*models.AddHost, 0, len(hosts))
+	for _, v := range hosts {
+		host := v.(map[string]interface{})
+		hs := models.AddHost{
+			DisplayName: core.StringPtr(host[Attr_DisplayName].(string)),
+			SysType:     core.StringPtr(host[Attr_SysType].(string)),
+		}
+		hostBody = append(hostBody, &hs)
+	}
+
+	body.Hosts = hostBody
+	body.HostGroupID = &hostgroupID
+	hostResponse, err := client.CreateHost(&body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	hostID := hostResponse[0].ID
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, hostID))
+	_, err = isWaitForIBMPIHostAvailable(ctx, client, hostID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceIBMPIHostRead(ctx, d, meta)
+}
+
+func resourceIBMPIHostRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, hostID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	client := instance.NewIBMPIHostGroupsClient(ctx, sess, cloudInstanceID)
+	host, err := client.GetHost(hostID)
+	if err != nil {
+		if strings.Contains(err.Error(), NotFound) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_HostID, host.ID)
+
+	capacity := []map[string]interface{}{}
+	if host.Capacity != nil {
+		cp, err := PIHostHostCapacityToMap(host.Capacity)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		capacity = append(capacity, cp)
+		d.Set(Attr_Capacity, capacity)
+	}
+	if host.DisplayName != "" {
+		d.Set(Attr_DisplayName, host.DisplayName)
+	}
+	if host.HostGroup != nil {
+		hostgroup, err := PIHostHostgroupToMap(host.HostGroup)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.Set(Attr_Hostgroup, hostgroup)
+	}
+	if host.State != "" {
+		d.Set(Attr_State, host.State)
+	}
+	if host.Status != "" {
+		d.Set(Attr_Status, host.Status)
+	}
+	if host.SysType != "" {
+		d.Set(Attr_SysType, host.SysType)
+	}
+
+	return nil
+}
+
+func resourceIBMPIHostUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, hostID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	displayName := d.Get(Arg_Host + ".0").(map[string]interface{})[Attr_DisplayName].(string)
```

Comment:
+ ".0" ?

Reply:
What is the question?

Reply:
I don't get why we are adding the .0 to the displayname when we get it.

Reply:
Getting the first element equivalent.